cache-filter/src/filter/http.rs

```diff
@@ -267,8 +307,9 @@ impl CacheFilter {
                     self.resume_http_request();
                     break;
                 }
-                Err(UpdateMetricsError::RateLimited) => {
+                Ok(RateLimitStatus::RateLimited(rate_limit_info)) => {
                     info!(self.context_id, "request is rate-limited");
+                    self.state.rate_limit_info = rate_limit_info;
                     self.send_http_response(429, vec![], Some(b"Request rate-limited.\n"));
```

Comment:
Here also we need to add rate limit headers. `send_http_response` uses local response which directly sends a response from here and destroy the filter object. `on_http_response_headers` gets triggered only  in the response path from upstream back to downstream.

Reply:
Actually Lahiru, this is not correct. I thought the same but after some testing, I saw `on_http_response_headers` is called even on the local response from `send_http_response`. You can even check the CI, in the rate-limit test, RateLimit headers are in fact sent back even though we are not adding it above. Interestingly, this fact allowed me to clean `visible-logs` up in the last PR as well ;)

Reply:
Ahh my bad !! It works because headers are added from the `on_http_response_headers` of the same filter. For filter chain `A -> B -> C`, if we do a local response from filter B, then the response path callbacks will get invoked from `B`  and downwards.